allennlp/modules/token_embedders/pretrained_transformer_embedder.py

```diff
@@ -101,6 +101,13 @@ def __init__(
             model_name,
             tokenizer_kwargs=tokenizer_kwargs,
         )
+
+        try:
+            self.transformer_model.resize_token_embeddings(len(tokenizer.tokenizer))
+        except NotImplementedError:
+            # Can't resize for transformers models that don't implement base_model.get_input_embeddings()
+            pass
```

Comment:
Is this completely safe? Maybe we should at least print a warning, because you have a good chance of running into trouble later, right?

Reply:
One of the unit tests in `pretrained_transformer_embedder_test.py` failed (`test_encoder_decoder_model`). Transformers lib call a `base_model.get_input_embeddings()`as part of the resize, but the actual base_model in an Encoder-Decoder model is the actual Encoder (BartEncoder, in the mentioned unit test), and they don't implement this:
https://github.com/huggingface/transformers/blob/1420b5ff675ccdc3296c6776b339a08a22d2e941/src/transformers/modeling_utils.py#L451. 
We could add a warning here, but I'm not sure which exact condition could be used here to trigger the warning. I _think_ that this exception could happen for other Encoder-Decoder models, but I'm not sure about all of them (nor I know them exaustively). I wouldn't be aware of this if it wasn't for the unit test that failed. 
What do you think, what do you propose?

Reply:
I would propose just a `logger.warning("Could not resize the token embedding matrix of the transformer model. This model does not support resizing.")`. But make sure to only print that message if the sizes actually differ!

If we're lucky, it will save someone hours down the line when they try to use this with BART and fail.

Reply:
OK, got it here. I thought there wouldn't be a single method for checking this, but there is...only the resizing that it seems to not being possible for encoder-decoders.